03_letters/CameronSchafer/letters_test.go

```diff
@@ -0,0 +1,88 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// main() test
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+	expected := "a:2\nb:1\n"
+	actual := buf.String()
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()\nexpected: %q\nactual: %q", expected, actual)
+	}
+}
+
+// sortLetters()
+func TestSortLetters(t *testing.T) {
+	// test cases with descriptions.
+	testCases := []struct {
+		description string
+		input       map[rune]int
+		expectedArr []string
+	}{
+		{description: "sortLetters [97:2, 98:1]", input: map[rune]int{97: 2, 98: 1},
+			expectedArr: []string{"a:2", "b:1"},
+		},
+		{description: "sortLetters [97:7, 98:7]", input: map[rune]int{97: 7, 98: 7},
+			expectedArr: []string{"a:7", "b:7"},
+		},
+		{description: "sortLetters [32:2, 97:1]", input: map[rune]int{32: 2, 97: 1},
+			expectedArr: []string{" :2", "a:1"},
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		// t.Run creates a sub test and runs it like a normal test
+		t.Run(test.description, func(t *testing.T) {
```

Comment:
Nice use of t.Run, I hadn't come across. I'm doing table-driven tests, but not using Run. I'll have to look at if for future labs ;)

Reply:
It is super useful if you need to run multiple tests through the same function